src/main/java/fr/inria/gforge/spoon/SpoonMojoGenerate.java

```diff
@@ -254,23 +254,18 @@ private void initSpoonProperties(Launcher launcher) throws MojoExecutionExceptio
 		}
 	}
 
-	private void addArtifactsInClasspathOfTargetClassLoader() throws IOException {
+	private URLClassLoader addArtifactsInClasspathOfTargetClassLoader() throws IOException {
+		URLClassLoader urlClassLoader = new URLClassLoader(new URL[0], getClass().getClassLoader());
 		// Changes classpath of the target class loader.
 		if (project.getArtifacts() == null || project.getArtifacts().isEmpty()) {
 			LogWrapper.info(this, "There is not artifact in this project.");
 		} else {
 			for (Artifact artifact : project.getArtifacts()) {
 				LogWrapper.debug(this, artifact.toString());
-				ClasspathHacker.addFile(artifact.getFile());
+				ClasspathHacker.addURL(urlClassLoader, artifact.getFile().toURI().toURL());
 			}
 		}
-
-		// Displays final classpath of the target classloader.
-		LogWrapper.info(this, "Running spoon with classpath:");
-		final URL[] urlClassLoader = urlsFromClassLoader(ClassLoader.getSystemClassLoader());
-		for (URL currentURL : urlClassLoader) {
-			LogWrapper.info(this, currentURL.toString());
-		}
```

Comment:
We probably could have kept those debug info no?

Reply:
the line above (264) already logs this.